### ISSUE TITLE
Go To Definition for whole Workspace?

### ISSUE THREAD

**Issue by vigh1000** [qingpeng9802/vscode-common-lisp#10]
**Is your feature request related to a problem? Please describe.**
Go To Definition works perfect when the functions were defined in the same file. Unfortunately this doesn't seem to work if the function was defined in another file. The file is present in a folder which is opened in VSCode as a workspace

**Describe the solution you'd like**
Go To Definition works also for files which are present in the same workspace. 

**Describe alternatives you've considered**
As a workaround i'm using the search in VSCode which works ok. But things like "Peek" etc. would be really neat to have :-).

**Additional context**
Just want to thank you for the plugin, I really like it :-)

Or am I'm missing something and Go To in Workspace should work?

**Comment by qingpeng9802** [comment]
The workspace support was considered. However, this support may be difficult to implement *correctly*.

Usually, when you need to use cross-file definition, you will either use `(load "../file.lisp")` or a system/package manager, for example `ASDF`. For `load` case, it might be easy to implement, but for `ASDF` case, things get complicated.

If this extension choose to process the ASDF information (kind of like compilation database), it is like re-implementing `ASDF` in TypeScript, which I would like to avoid. If you need `ASDF` things, I would suggest to try [Common Lisp
 ailisp](https://marketplace.visualstudio.com/items?itemName=ailisp.commonlisp-vscode) or [Alive
 Rich Heller](https://marketplace.visualstudio.com/items?itemName=rheller.alive). These extensions use the symbols from lisp directly, and they might work with `ASDF` (although I did not confirm).

In short, `load(` support is possible to be a feature, but `ASDF` support is more like a design limitation.

**Comment by vigh1000** [comment]
Thank you for the quick answer!
We're using `(load "../file.lisp")`. So I've got a bunch of files in my workspace where the functions are defined and using them in different files. 
So if "Go To Definition" would also consider these files, I would be happy :-) 

But this is probably what you mean with "implement it _correctly_"?

**Comment by qingpeng9802** [comment]
`ASDF` way is a de facto standard for most development so what I meant was that correctly implementing a fake `ASDF`.

For `(load` case, the feature might not be available very soon. The design of the buffer mechanism for files is needed to be considered carefully.
